### ISSUE TITLE
Unable to link with VSCode/PlatformIO

### ISSUE THREAD

**Issue by obones** [boblemaire/esp32HTTPrequest#5]
I'm using this library with VSCode/PlatformIO targetting the `espressif32` platform and while the compilation goes fine, I'm getting this error from `ld`:

```
.pio\build\genericESP32\lib1fe\libesp32HTTPrequest.a(esp32HTTPrequest.cpp.o):(.literal._ZN16esp32HTTPrequest5_sendEPKcj+0xc): 
undefined reference to `TLSlock_S'
```

From the name, it seems related to TLS, but I can't figure out what's missing in my configuration. Should I activate a specific option for TLS support in my target?

**Comment by obones** [comment]
Ok, I should have thought about it earlier, but searching for `TLSlock_S` in all sources indicated that it is declared in `esp32HTTPrequest.h` like this:

    extern SemaphoreHandle_t TLSlock_S;     // Mutex to limit to one HTTPS active

But it is nowhere to be seen in `esp32HTTPrequest.cpp`.
I thus added it at the end of that file, like this:

    SemaphoreHandle_t TLSlock_S = nullptr; 

It now links, but frankly, I'm not sure this is enough. I mean, it's never allocated a value received from `xSemaphoreCreateMutex` which I would have expected to find.

**Comment by boblemaire** [comment]
Yeah, that's a little sloppy.  This conversion of the original asyncHTTPrequest for ESP8266 has morphed into something quite different.  These are both clients that I use in the IoTaWatt firmware.  With the ESP8266 I really needed to have an async capability, and it works great there.

For ESP32, the async capability isn't really necessary if you run it in a separate FREErtos task.  Moreover, I could never get the underlying async feature of the ESP32 HTTP to work anyway.  What I did need in the ESP32 was the ability to do HTTPS, and so that is what this does.  I guess I should have renamed it to say that.

So this client will do HTTPS (TLS).  However I have run into problems when I allow two or more TLS transactions to run at the same time.  My code can live with serializing the TLS requests, so I put in the semaphore to do so.  But I guess I crossed the line by declaring the semaphore elsewhere in my code.  I suspect you are getting away with the null semaphore pointer because you are not using TLS (specifying a URL with https://).

I'm about to work on the ESP32 port again in the next few weeks.  I'll clean that up.
